### PR TITLE
[Component with its own class name as service type] LPS-199542 Refactor Helper components from document-library, document-library-opener-google-drive-web, on-demand-admin, portal-workflow and user-associated-data-web modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Previous pull: https://github.com/liferay-core-infra/liferay-portal/pull/5933

Ticket: https://liferay.atlassian.net/browse/LPS-199542

Hey @tinatian I tried to merge the two methods in BasePortletToolbarContributor, but this https://github.com/liferay-core-infra/liferay-portal/pull/5933#issuecomment-1809676521 happened, because the logic of knowing if the show actions is enabled or if a folder does not exists was conflicting.

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:lima

**Comment by MrJohn1911** [comment]
ci:test:um

**Comment by MrJohn1911** [comment]
ci:test:usm

**Comment by MrJohn1911** [comment]
ci:test:workflow

**Comment by Kvale1733** [comment]
ci:test:lima

**Comment by catarineaguiar** [comment]
ci:test:dl-store

**Comment by catarineaguiar** [comment]
no regressions found

**Review by rafaprax — APPROVED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:lima

**Comment by catarineaguiar** [comment]
no regression found

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/BasePortletToolbarContributor.java [hunk 1]
```diff
@@ -98,9 +99,7 @@ public Folder getFolder(
 		return folder;
 	}
 
-	public Boolean isShowActionsEnabled(
-		ThemeDisplay themeDisplay, PortletRequest portletRequest) {
-
+	protected Boolean isShowActionsEnabled(ThemeDisplay themeDisplay) {
```
Comment (dantewang) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/BasePortletToolbarContributor.java:102]: @MrJohn1911 This method doesn't need to return a `Boolean`, as `boolean` is enough. Please change it, thanks.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found.

**Comment by MrJohn1911** [comment]
ci:test:usm

**Comment by MrJohn1911** [comment]
ci:test:stable

**Comment by zhangyan0701** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:um

**Comment by zhangyan0701** [comment]
ci:test:lima

**Comment by zhangyan0701** [comment]
ci:test:workflow

**Comment by zhangyan0701** [comment]
ci:test:headless

**Comment by zhangyan0701** [comment]
ci:test:object

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by tinatian** [comment]
@MrJohn1911 Please follow the same pattern of https://github.com/liferay-core-infra/liferay-portal/pull/6147/commits/38a5f73bbfd2ddcbe259cf7cc21d7c4751786e32 to avoid using ***Util, this way, it would be clear to show the dependency chain by @Reference.

Please resend.
Thanks.
Tina.

**Pull request closed without merging**